dpnp/dpnp_array.py

```diff
@@ -1000,15 +993,61 @@ def take(self, indices, axis=None, out=None, mode='raise'):
 
     def transpose(self, *axes):
         """
-        Returns a view of the array with axes permuted.
+        Returns a view of the array with axes transposed.
 
-        .. seealso::
-           :obj:`dpnp.transpose` for full documentation,
-           :meth:`numpy.ndarray.reshape`
+        For full documentation refer to :obj:`numpy.ndarray.transpose`.
+
+        Returns
+        -------
+        y : dpnp.ndarray
+            View of the array with its axes suitably permuted.
 
+        See Also
+        --------
+            :obj:`dpnp.transpose` : Equivalent function.
+            :obj:`dpnp.ndarray.ndarray.T` : Array property returning the array transposed.
+            :obj:`dpnp.ndarray.reshape` : Give a new shape to an array without changing its data.
+
+        Examples
+        --------
+        >>> import dpnp as dp
+        >>> a = dp.array([[1, 2], [3, 4]])
+        >>> a
+        array([[1, 2],
+               [3, 4]])
+        >>> a.transpose()
+        array([[1, 3],
+               [2, 4]])
+        >>> a.transpose((1, 0))
+        array([[1, 3],
+               [2, 4]])
+        >>> a.transpose(1, 0)
+        array([[1, 3],
+               [2, 4]])
+
+        >>> a = dp.array([1, 2, 3, 4])
+        >>> a
+        array([1, 2, 3, 4])
+        >>> a.transpose()
+        array([1, 2, 3, 4])
+            
         """
 
-        return dpnp.transpose(self, axes)
+        ndim = self.ndim
+        if ndim < 2:
+            return self
+
+        res = self.__new__(dpnp_array)
+        if ndim == 2:
```

Comment:
axes is not used when `ndim==2`, will provide different results compared to numpy
`b_dpnp = dpnp.array([[1, 2], [3, 4]])`
`b_numpy = numpy.array([[1, 2], [3, 4]])`

`numpy.transpose(b_numpy, (0,1))`
`array([[1, 2],       [3, 4]])`

`dpnp.transpose(b_dpnp, (0,1))`
`array([[1, 3],       [2, 4]])`

Reply:
Fixed by new commit, thanks!